src/header-bidding/prebid/improve-digital.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { createAdSize } from 'core';
+import {
+	isInAuOrNz as isInAuOrNz_,
+	isInRow as isInRow_,
+	isInUk as isInUk_,
+	isInUsOrCa as isInUsOrCa_,
+} from 'utils/geo-utils';
+import { getBreakpointKey as getBreakpointKey_ } from '../utils';
+import {
+	getImprovePlacementId,
+	getImproveSkinPlacementId,
+} from './improve-digital';
+
+const getBreakpointKey = getBreakpointKey_ as jest.Mock;
+const isInAuOrNz = isInAuOrNz_ as jest.Mock;
+const isInRow = isInRow_ as jest.Mock;
+const isInUk = isInUk_ as jest.Mock;
+const isInUsOrCa = isInUsOrCa_ as jest.Mock;
+
+jest.mock('experiments/ab', () => ({
+	isInVariantSynchronous: jest.fn(),
+}));
+jest.mock('utils/geo-utils');
+
+jest.mock('../utils', () => ({
+	...jest.requireActual('../utils'),
+	getBreakpointKey: jest.fn(),
+}));
+
+describe('getImprovePlacementId', () => {
+	afterEach(() => {
+		jest.resetAllMocks();
+	});
+
+	test.each([
+		{ sizes: [createAdSize(970, 250)], expectedId: 22987847 },
+		{ sizes: [createAdSize(728, 90)], expectedId: 1116397 },
+		{
+			sizes: [createAdSize(728, 90), createAdSize(970, 250)],
+			expectedId: 1116397,
+		},
+		{ sizes: [createAdSize(300, 250)], expectedId: 1116396 },
+		{ sizes: [createAdSize(300, 600)], expectedId: 1116396 },
+		{ sizes: [createAdSize(1, 2)], expectedId: -1 },
+	])(
+		'in UK on desktop case %p gives correct placement id',
+		({ sizes, expectedId }) => {
+			isInUk.mockReturnValue(true);
+			getBreakpointKey.mockReturnValueOnce('D');
+			expect(getImprovePlacementId(sizes)).toBe(expectedId);
+		},
+	);
+
+	test.each([
+		{ sizes: [createAdSize(970, 250)], expectedId: 1116399 },
+		{ sizes: [createAdSize(728, 90)], expectedId: 1116399 },
+		{
+			sizes: [createAdSize(728, 90), createAdSize(970, 250)],
+			expectedId: 1116399,
+		},
+		{ sizes: [createAdSize(300, 250)], expectedId: 1116398 },
+		{ sizes: [createAdSize(300, 600)], expectedId: 1116398 },
+		{ sizes: [createAdSize(1, 2)], expectedId: -1 },
+	])(
+		'in UK and on tablet %p returns correct placement id',
+		({ sizes, expectedId }) => {
+			isInUk.mockReturnValue(true);
+			getBreakpointKey.mockReturnValueOnce('T');
+			expect(getImprovePlacementId(sizes)).toBe(expectedId);
+		},
+	);
+
+	test.each([
+		{ sizes: [createAdSize(970, 250)], expectedId: -1 },
+		{ sizes: [createAdSize(728, 90)], expectedId: -1 },
+		{
+			sizes: [createAdSize(728, 90), createAdSize(970, 250)],
+			expectedId: -1,
+		},
+		{ sizes: [createAdSize(300, 250)], expectedId: 1116400 },
+		{ sizes: [createAdSize(300, 600)], expectedId: 1116400 },
+		{ sizes: [createAdSize(1, 2)], expectedId: -1 },
+	])(
+		'in UK and on M %p returns correct placement id',
```

Comment:
on mobile?

Reply:
Glad you're paying attention! Have included those changes now